core/server/api/mail.js

```diff
@@ -11,16 +10,14 @@ var Promise       = require('bluebird'),
     notifications = require('./notifications'),
     i18n          = require('../i18n'),
     docName       = 'mail',
-    mode          = config.get('env'),
-    testing       = mode !== 'production' && mode !== 'development',
     mailer,
     apiMail;
 
 /**
  * Send mail helper
  */
 function sendMail(object) {
-    if (!(mailer instanceof mail.GhostMailer) || testing) {
```

Comment:
Solved by changing the test, see https://github.com/TryGhost/Ghost/pull/7929/files#diff-0387d2aac26fc08f7e8d00afe8acca6eR25